.github/workflows/publish.yml

```diff
@@ -0,0 +1,88 @@
+name: Publish
+
+on:
+  workflow_call
+
+jobs: 
+
+  publish-start-notification:
+    name: 'Publish Start Notification'
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: ./.github/workflows/slackNotification.yml
+        with:
+          title: "Publish waiting for approval in github."
+          notification: "Main has been merged and publish job is ready for approval."
+          type: "notification"
+          workflow: "publish.yml"
+
+  publish-approved-slack: 
+    runs-on: 'ubuntu-latest'
+    environment: publish
```

Comment:
This environment will have an approval in the github action configuration.  This is how we pause for github actions. 

Note the flow is 
Tests, Build All  -> publish -> slack notification -> wait for approval. 

Where we use to have two approvals for a publish we now have one. The only reason I kept that approval is it seems like a good step to install and verify the built vsix files before shipping it to microsoft

Reply:
agreed..def view this as a first pass and we can continue to improve as we go along